tests/integration/hive_metastore/test_table_move.py

```diff
@@ -0,0 +1,103 @@
+    from_catalog = make_catalog()
+    from_schema = make_random(4)
+    to_catalog = make_catalog()
+    tm = TableMove(ws, sql_backend)
+    tm.move_tables(from_catalog.name, from_schema, "*", to_catalog.name, from_schema, False)
+    assert (
+        len(
+            [
+                rec.message
+                for rec in caplog.records
+                if f"schema {from_schema} not found in catalog {from_catalog.name}" in rec.message
+            ]
+        )
+        == 1
+    )
+
+
+@retried(on=[NotFound], timeout=timedelta(minutes=2))
+def test_move_tables(ws, sql_backend, make_catalog, make_schema, make_table, make_acc_group):
+    tm = TableMove(ws, sql_backend)
+    group_a = make_acc_group()
+    group_b = make_acc_group()
+    from_catalog = make_catalog()
+    from_schema = make_schema(catalog_name=from_catalog.name)
+    from_table_1 = make_table(catalog_name=from_catalog.name, schema_name=from_schema.name)
+    from_table_2 = make_table(catalog_name=from_catalog.name, schema_name=from_schema.name)
+    from_table_3 = make_table(catalog_name=from_catalog.name, schema_name=from_schema.name)
+    from_view_1 = make_table(
+        catalog_name=from_catalog.name,
+        schema_name=from_schema.name,
+        view=True,
+        ctas=f"select * from {from_table_2.full_name}",
+    )
+    to_catalog = make_catalog()
+    to_schema = make_schema(catalog_name=to_catalog.name)
+    # creating a table in target schema to test skipping
+    to_table_3 = make_table(catalog_name=to_catalog.name, schema_name=to_schema.name, name=from_table_3.name)
+    sql_backend.execute(f"GRANT SELECT ON TABLE {from_table_1.full_name} TO `{group_a.display_name}`")
+    sql_backend.execute(f"GRANT SELECT,MODIFY ON TABLE {from_table_2.full_name} TO `{group_b.display_name}`")
+    sql_backend.execute(f"GRANT SELECT ON VIEW {from_view_1.full_name} TO `{group_b.display_name}`")
+    sql_backend.execute(f"GRANT SELECT ON TABLE {to_table_3.full_name} TO `{group_a.display_name}`")
+    tm.move_tables(from_catalog.name, from_schema.name, "*", to_catalog.name, to_schema.name, False)
+    tables = ws.tables.list(catalog_name=to_catalog.name, schema_name=to_schema.name)
+    table_1_grant = ws.grants.get(
+        securable_type=SecurableType.TABLE, full_name=f"{to_catalog.name}.{to_schema.name}.{from_table_1.name}"
+    )
+    table_2_grant = ws.grants.get(
+        securable_type=SecurableType.TABLE, full_name=f"{to_catalog.name}.{to_schema.name}.{from_table_2.name}"
+    )
+    table_3_grant = ws.grants.get(
+        securable_type=SecurableType.TABLE, full_name=f"{to_catalog.name}.{to_schema.name}.{from_table_3.name}"
+    )
+    view_1_grant = ws.grants.get(
+        securable_type=SecurableType.TABLE, full_name=f"{to_catalog.name}.{to_schema.name}.{from_view_1.name}"
+    )
+    assert (
+        len(
+            [t for t in tables if t.name in [from_table_1.name, from_table_2.name, from_table_3.name, from_view_1.name]]
+        )
+        == 4
+    )
```

Comment:
this is unreadable, please refactor. don't assert for length - use "assert key in dict" instead